src/store-api/src/region_request.rs

```diff
@@ -501,6 +515,60 @@ impl TryFrom<v1::AddColumnLocation> for AddColumnLocation {
     }
 }
 
+/// Modify a column.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ModifyColumn {
+    /// Schema of the column to modify.
+    pub column_name: String,
+    /// Column will be modified to this type.
+    pub target_type: ConcreteDataType,
+}
+
+impl ModifyColumn {
+    /// Returns an error if the column to modify is invalid.
+    pub fn validate(&self, metadata: &RegionMetadata) -> Result<()> {
+        let column_meta =
+            metadata
+                .column_by_name(&self.column_name)
+                .context(InvalidRegionRequestSnafu {
+                    region_id: metadata.region_id,
+                    err: format!("column {} not found", self.column_name),
+                })?;
+
+        ensure!(
+            !matches!(
+                column_meta.semantic_type,
+                SemanticType::Timestamp | SemanticType::Tag
+            ),
+            InvalidRegionRequestSnafu {
+                region_id: metadata.region_id,
+                err: "'timestamp' or 'tag' column cannot change type".to_string()
+            }
+        );
+        ensure!(
+            column_meta
+                .column_schema
+                .data_type
+                .can_arrow_type_cast_to(&self.target_type),
+            InvalidRegionRequestSnafu {
+                region_id: metadata.region_id,
+                err: format!(
+                    "column '{}' cannot be cast automatically to type '{}'",
+                    self.column_name, self.target_type
+                ),
+            }
+        );
+
+        Ok(())
+    }
+
+    /// Returns true if no column to modify to the region.
```

Comment:
Looks like the comment is incorrect?

Reply:
> Looks like the comment is incorrect?

yep, actually I have reported this to @waynexia before, and I intentionally kept this issue in order to maintain uniformity.